aten/src/ATen/WrapDimUtils.h

```diff
@@ -85,7 +85,7 @@ inline int64_t _legacy_cat_wrap_dim(
     int64_t dim,
     const std::vector<std::vector<T>>& tensor_sizes) {
   for (auto& sizes : tensor_sizes) {
-    if (sizes == std::vector<T>({0})) {
```

Comment:
@Chillee here's what's going on:

sizes is {SymInt(0)}
std::vector<T>({0}) is actually {} for reasons i am still trying to understand.

so the previous code never did the 'continue' it wanted to do, but by explicitly checking .size() and sizes[0] it does.